pkg/kubelet/cm/cpumanager/topology_hints.go

```diff
@@ -0,0 +1,124 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cpumanager
+
+import (
+	"k8s.io/api/core/v1"
+	"k8s.io/klog"
+	"math"
+
+	"k8s.io/kubernetes/pkg/kubelet/cm/cpumanager/topology"
+	"k8s.io/kubernetes/pkg/kubelet/cm/cpuset"
+	"k8s.io/kubernetes/pkg/kubelet/cm/topologymanager"
+	"k8s.io/kubernetes/pkg/kubelet/cm/topologymanager/socketmask"
+)
+
+func (m *manager) GetTopologyHints(pod v1.Pod, container v1.Container) ([]topologymanager.TopologyHint, bool) {
```

Comment:
This doesn't seem to have been updated according to my suggestions in: https://docs.google.com/document/d/1UbD1igeBA1sRuqwkQgJ5sJ9sf1rao_LLORWLhwSdGUo

Notes copied from the doc:
```
Checkpointing TopologyHints
As a side note, I had to add some extra logic to the GetTopologyHints() implementation
for the cpumanager to get it to work across kubelet restarts. The main problem being that
the topology information was lost across the restart (since it’s not being checkpointed) and
new hints were being generated for already-running containers. We could either checkpoint
the hint information, or make sure we regenerate valid hints as I do below...
```

Reply:
Done - may need editing as implemented from notes?

Reply:
It says this comment is outdated, but I don't see the code change in the file.